src/platform_impl/macos/observer.rs

```diff
@@ -169,31 +179,50 @@ impl Default for EventLoopWaker {
                 ptr::null_mut(),
             );
             CFRunLoopAddTimer(CFRunLoopGetMain(), timer, kCFRunLoopCommonModes);
-            EventLoopWaker { timer }
+            EventLoopWaker {
+                timer,
+                start_instant: Instant::now() - Duration::from_secs(1000),
+                next_fire_date: None,
+            }
         }
     }
 }
 
 impl EventLoopWaker {
     pub fn stop(&mut self) {
-        unsafe { CFRunLoopTimerSetNextFireDate(self.timer, std::f64::MAX) }
+        if self.next_fire_date.is_some() {
+            self.next_fire_date = None;
```

Comment:
Could we `take` inside the `if`?

Reply:
yeah, technically, but that looks weird to me because the logic looks back to front - like `.take().is_some()`.

The intent is that we only want to change state if `.is_some()` and it now it looks like we're changing state unconditionally and checking what we've done before deciding whether to change more state.